### ISSUE TITLE
remark-lint-no-unused-definitions gives false positives when used within html

### ISSUE THREAD

**Issue by ppiyush13** [remarkjs/remark-lint#218]
`remark-lint-no-unused-definitions` is giving false positives when link is used within html tags. For example:

```
<div align="center">
     [react link][react]
</div> 

[react]: https://github.com/facebook/react
```

Though react is used on second line, still it gives warning as found unused definition [react]

In actual it should not give any warning

**Comment by wooorm** [comment]
The `[react link][react]` part of your file, is not a reference. It is part of HTML. According to CommonMark, `<div ...> starts an HTML block, that is closed by a blank line. Therefore, it is closed right before the definition, making the value that looks like a reference, raw HTML.

You can try it out in the [CommonMark dingus](https://spec.commonmark.org/dingus/).

This warning is correct.

**Comment by ppiyush13** [comment]
@wooorm , thank you for the clarification. It was really helpful.

**Comment by wooorm** [comment]
Alright, I’ll close this then! Thanks for using remark-lint and for raising an issue!
